sdk/inc/azure/iot/az_iot_hub_client.h

```diff
@@ -43,6 +43,12 @@ typedef struct
 {
   /**
    * The module name (if a module identity is used).
+   * If the module name contains any of the following characters,
```

Comment:
I think this isn't describing the whole problem. For example, devices could contain Unicode higher-plane characters or be characters interpreted in non-Latin alphabets.

All parts of the Topic (deviceid, moduleid, user-agent, properties, etc) _must_ follow the guidelines regarding MQTT 3.1.1 (and future versions) topic names described in the Oasis MQTT spec. E.g. [MQTT 3.1.1 Section 4.7 Topic Names](http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc398718106).

Device and module IDs that contain characters unallowed by the MQTT specification must use URI escaping as described by [RFC 3986, section 2.1](https://datatracker.ietf.org/doc/html/rfc3986/#section-2.1)